src/object-slice.js

```diff
@@ -3,43 +3,33 @@ const ArraySlice = require('./array-slice');
 
 class ObjectSlice extends ArraySlice {
   map(callback, thisArg) {
-    return this.elements.map(function (member) {
-      return callback(member.value, member.key, member);
-    }, thisArg);
+    return this.elements.map(member => callback(member.value, member.key, member), thisArg);
   }
 
   filter(callback, thisArg) {
-    return new ObjectSlice(this.elements.filter(function (member) {
-      return callback(member.value, member.key, member);
-    }, thisArg));
+    return new ObjectSlice(this.elements.filter(member => callback(member.value, member.key, member), thisArg));
   }
 
   reject(callback, thisArg) {
     return this.filter(negate(callback, thisArg));
   }
 
   forEach(callback, thisArg) {
-    return this.elements.forEach(function (member, index) {
-      return callback(member.value, member.key, member, index);
-    }, thisArg);
+    return this.elements.forEach((member, index) => callback(member.value, member.key, member, index), thisArg);
```

Comment:
Wondering whether the fat arrow won't collide with the `thisArg`. Do we have this behavior tested so we're sure it works as intended even after this change?

(Relevant on multiple places in this commit)

Reply:
I naively though that would've been tested, but I can see that probably isn't the case. I'll test this behaviour and write missing tests.

Reply:
Addressed in db73cc4.